src/components/TresCanvas.vue

```diff
@@ -131,17 +109,45 @@ onMounted(() => {
         immediate: true
     })
 
-    if (!camera.value) {
+})
+
+const addDefaultCamera = () => {
+    const camera = new PerspectiveCamera(45, window.innerWidth / window.innerHeight, 0.1, 1000)
+    camera.position.set(3, 3, 3)
+    camera.lookAt(0, 0, 0)
+    context.addCamera(camera)
+
+    if (context.cameras.value.length >= 2) {
+        camera.removeFromParent()
+        context.removeCamera(camera)
+    }
+}
+
+const unwatch = watch(canvas, val => {
+    if (!val) return
+    triggerRef(scene)
+    unwatch?.()
+})
+
+watch(scene, () => {
+    if (!context.camera.value) {
         logWarning(
             'No camera found. Creating a default perspective camera. ' +
-            'To have full control over a camera, please add one to the scene.'
+            'To have full control over a camera, please add one to the scene.',
         )
         addDefaultCamera()
     }
-
-    if (import.meta.hot)
-        import.meta.hot.on('vite:afterUpdate', dispose)
 })
```

Comment:
to void this warning is always shown, cuz this run before the camera is assigned. now check only run once by `triggerRef`